src/tls_relation.py

```diff
@@ -85,7 +107,9 @@ def certificate_relation_created(self, hostname: str) -> None:
             hostname: Certificate's hostname.
         """
         # At this point, TLSInformation state component should already be initialized
-        tls_integration = typing.cast(Relation, self.model.get_relation(self.integration_name))
+        tls_integration = self.model.get_relation(self.integration_name)
+        assert tls_integration
```

Comment:
this will probably lead to a bandit warning once linting is enabled: https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html

Reply:
We can use `raise AssertionError` instead

Reply:
Thanks, I updated the code to raise AssertionError.